### DISCUSSION TITLE
Does db.session.delete() delete associated data?

### DISCUSSION THREAD

**Discussion by DAAworld** [apache/superset#17789] [Q&A / Help]
Does db.session.delete() delete associated data? I want to change the physical delete to the logical delete using the "is_del" field distinction, but I am afraid that the associated data will not change according to the original logic, causing the data that should be logically deleted to be found

**Comment by nytai** [comment]
It would only delete the associated data if the associations were defined as `on delete cascade`, so the answer is unfortunately it depends on the implementation.
